addons/mail/static/src/models/messaging.js

```diff
@@ -329,6 +317,26 @@ registerModel({
                 this.env.services.bus_service.forceUpdateChannels();
             }
         },
+        /**
+         * @private
+         */
+        _onChangeAllPersonas() {
+            const partnerIds = [];
+            const guestIds = [];
+            for (const persona of this.allPersonas) {
+                if (persona.partner) {
+                    partnerIds.push(persona.partner.id);
+                } else {
+                    guestIds.push(persona.guest.id);
+                }
+            }
+            if (partnerIds.length) {
+                this.env.services['im_status'].registerToImStatus('res.partner', partnerIds);
+            }
+            if (guestIds.length) {
+                this.env.services['im_status'].registerToImStatus('mail.guest', guestIds);
+            }
+        },
```

Comment:
This `onChange` can be frequently invoked, while it's execution is not blazing fast.

A solution would be to use `Throttle` so it does processing and makes register every X (milli)seconds.
Something like:
```js
// messaging.js
_onChangeAllPersonas() {
    this.update({
        updateImStatusRegisterThrottle: { func: this.registerToImStatus }
     });
}
updateImStatusRegisterThrottle: one('Throttle', {
    inverse: 'messagingAsUpdateImStatusRegister',
}),
// throttle.js
messagingAsUpdateImStatusRegister: one('Messaging', {
    identifying: true,
    inverse: 'updateImStatusRegisterThrottle',
}),
duration: {
    compute() {
        if(this.messagingAsUpdateImStatusRegister) {
            return 10 * 1000; // 10 seconds
        }
    }
}
```

That way, the onChange is immediate, and do the processing + register only once per 10 seconds at most.